### PR TITLE
[AMD] Add build and tests for ROCm 7.2 nightly images

### PR DESCRIPTION
**Pull request by akao-amd**
## Motivation

Enable people access SGLang with the latest official ROCm 7.2 stack.  Active testing is driven in #17799 .

## Modifications

* Add the build of two ROCm 7.2 based images in the nightly workflow.
* Rebase the docker environment: from vllm to rocm/pytorch.
* Hot fixes are added to enable DeepSeek-R1-MXFP4 and DeepSeek-V3.2.
* Known issues
  * GPT-OSS-20B/120B: need later AITER support.
  * vllm dependency is not yet clean up in the scope of unit tests. 

## Accuracy Tests

Check [the log](https://github.com/sgl-project/sglang/actions/runs/21468531934) for the current best result. In terms of accuracy, there is no regression found in the preview image so far for more than 20 models.

## Benchmarking and Profiling

Same as above.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: docker/rocm.Dockerfile [hunk 1]
```diff
@@ -92,11 +117,34 @@ RUN if [ "$BUILD_LLVM" = "1" ]; then \
...
+      *rocm720*) \
+        echo "ROCm 7.2 flavor detected from GPU_ARCH=${GPU_ARCH}"; \
+        cd aiter \
+        && sed -i '290 s/64/128/' csrc/py_itfs_cu/asm_mla.cu \
+        && sed -i '292 s/qh64/qh16/' csrc/py_itfs_cu/asm_mla.cu \
+        && sed -i '293 s/qh64/qh16/' csrc/py_itfs_cu/asm_mla.cu \
+        && sed -i '326 s/64/128/' csrc/py_itfs_cu/asm_mla.cu \
+        && sed -i '328 s/qh64/qh16/' csrc/py_itfs_cu/asm_mla.cu \
+        && sed -i '329 s/qh64/qh16/' csrc/py_itfs_cu/asm_mla.cu \
+        && sed -i '460 s/if.*:/if False:/' aiter/ops/triton/pa_mqa_logits.py; \
```
Comment (hubertlu-tw) [inline docker/rocm.Dockerfile:151]: For reference, the issue is being tracked by https://github.com/ROCm/triton-internal/issues/1602
